test_programs/compile_failure/type_annotation_needed_on_struct_constructor/src/main.nr

```diff
@@ -0,0 +1,6 @@
+struct Foo<T> {
+}
+
+fn main() {
+    let foo = Foo {};
+}
```

Comment:
Neither integration test uses the stdlib so let's move them to the frontend tests

Reply:
Sounds good. I think I put them here because before this PR, these programs would error during monomorphization and those errors aren't caught in the frontend tests. But... now that they do error, I guess they make sense there.

Reply:
Ah, we still catch these during monomorphization, right. May need to keep these here then.